File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/webapp/WebAppSetting.js
define([ "stjs" ], function(stjs) {
	var WebAppSetting = function(name, value) {
		this.setName(name);
		this.setValue(value);
	};
	stjs.extend(WebAppSetting, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.webapp.WebAppSetting';
		constructor.serialVersionUID = 1;
		prototype.name = null;
		prototype.value = null;

		prototype.getName = function() {
			return this.name;
		};
		prototype.setName = function(name) {
			this.name = name;
		};

		prototype.getValue = function() {
			return this.value;
		};
		prototype.setValue = function(value) {
			this.value = value;
		};
	}, {});
	return WebAppSetting;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/webapp/WebAppSetting.java
/*
 * Copyright 2017 ETH Zuerich, SIS
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.webapp;

import java.io.Serializable;

import org.apache.commons.lang3.builder.ToStringBuilder;
import org.apache.commons.lang3.builder.ToStringStyle;

import com.fasterxml.jackson.annotation.JsonProperty;

import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("as.dto.webapp.WebAppSetting")
public class WebAppSetting implements Serializable
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private String name;

    @JsonProperty
    private String value;

    public WebAppSetting()
    {
    }

    public WebAppSetting(String name, String value)
    {
        setName(name);
        setValue(value);
    }

    public String getName()
    {
        return name;
    }

    public void setName(String name)
    {
        this.name = name;
    }

    public String getValue()
    {
        return value;
    }

    public void setValue(String value)
    {
        this.value = value;
    }

    @Override
    public String toString()
    {
        ToStringBuilder builder = new ToStringBuilder(this, ToStringStyle.SHORT_PREFIX_STYLE);
        builder.append(name);
        builder.append(value);
        return builder.toString();
    }

}
